Create `yotpo_rich_snippets` table if absent during module upgrade

fredden opened the pull request:
Fixes: #144

From what I can tell this table was introduced in version 2.4 in an install schema script (see 02b6571e), but the upgrade schema was never updated to match. This means that if a version prior to v2.4 was installed, the required table is absent. Upgrading to version 3.1.0 or later produces an error due to 2f2a3a2c assuming that the table exists. This pull request resolves that problem.

fredden committed:
Install table if absent during module upgrade

pniel-cohen reviewed (changes requested):
That's fine excepts that this table should be installed using the default connection & not the `$salesConnection`.

fredden committed:
Use $defaultConnection for Rich Snippets table

pniel-cohen reviewed (commented):
Just by looking, it seems fine to me.
*Still requires relevant QA before merging!

sharp80 commented on Setup/UpgradeSchema.php:
```diff
@@ -220,10 +220,53 @@ public function upgrade(SchemaSetupInterface $setup, ModuleContextInterface $con
         }
 
-        $richSnippetsTable = $installer->getConnection()->describeTable($installer->getTable('yotpo_rich_snippets'));
+        if (!$defaultConnection->isTableExists($yotpoRichSnippetsTable)) {
```
we already have the code of creating this table - we need to make sure we don't duplicate this code.

fredden replied:
Hi sharp80. Did you get a chance to read through the pull request description and linked issue? I don't think I understand your comment here. There is code to create the table upon module installation, but not on upgrade. Are you asking for a code change?

sharp80 replied:
@fredden - i meant that we have the same code that creates richSnippets table in a different file (InstallScheme.php) and better practice is to move the actual code that creates the table to utility function and call it from the upgradeScheme and from installScheme.

pniel-cohen replied:
What @fredden meant was:
The current installation code only runs on installation & not on upgrade.
On an extremely rare situation, when someone had a very old version of the yotpo module installed & now upgrading to a newer version, the table won't be created.
So it's not a bad idea to put that on the upgrade script as well, just i case.
Also, I think we should migrate to declarable scheme anyway soon, so it won't matter.

fredden replied:
Thanks for explaining. I've removed the duplicate code.

fredden committed:
Remove duplicate database table create code

UpgradeSchema will run on install as well as upgrade, and this already contains
code to create these two tables if they do not already exist.

fredden committed:
Merge remote-tracking branch 'upstream/master' into setup/upgrade-schema

lgreenboim commented:
Thank you fo the fix, it’ll be included in the next release - 3.1.3
